\documentclass[journal]{vgtc}                % final (journal style)
%\documentclass[review,journal]{vgtc}         % review (journal style)
%\documentclass[widereview]{vgtc}             % wide-spaced review
%\documentclass[preprint,journal]{vgtc}       % preprint (journal style)

%% Uncomment one of the lines above depending on where your paper is
%% in the conference process. ``review'' and ``widereview'' are for review
%% submission, ``preprint'' is for pre-publication, and the final version
%% doesn't use a specific qualifier.

%% Please use one of the ``review'' options in combination with the
%% assigned online id (see below) ONLY if your paper uses a double blind
%% review process. Some conferences, like IEEE Vis and InfoVis, have NOT
%% in the past.

%% Please note that the use of figures other than the optional teaser is not permitted on the first page
%% of the journal version.  Figures should begin on the second page and be
%% in CMYK or Grey scale format, otherwise, colour shifting may occur
%% during the printing process.  Papers submitted with figures other than the optional teaser on the
%% first page will be refused.

%% These few lines make a distinction between latex and pdflatex calls and they
%% bring in essential packages for graphics and font handling.
%% Note that due to the \DeclareGraphicsExtensions{} call it is no longer necessary
%% to provide the the path and extension of a graphics file:
%% \includegraphics{diamondrule} is completely sufficient.
%%
\ifpdf%                                % if we use pdflatex
  \pdfoutput=1\relax                   % create PDFs from pdfLaTeX
  \pdfcompresslevel=9                  % PDF Compression
  \pdfoptionpdfminorversion=7          % create PDF 1.7
  \ExecuteOptions{pdftex}
  \usepackage{graphicx}                % allow us to embed graphics files
  \DeclareGraphicsExtensions{.pdf,.png,.jpg,.jpeg} % for pdflatex we expect .pdf, .png, or .jpg files
\else%                                 % else we use pure latex
  \ExecuteOptions{dvips}
  \usepackage{graphicx}                % allow us to embed graphics files
  \DeclareGraphicsExtensions{.eps}     % for pure latex we expect eps files
\fi%

%% it is recomended to use ``\autoref{sec:bla}'' instead of ``Fig.~\ref{sec:bla}''
\graphicspath{{figures/}{pictures/}{images/}{./}} % where to search for the images

\usepackage{color}
\usepackage{microtype}                 % use micro-typography (slightly more compact, better to read)
\PassOptionsToPackage{warn}{textcomp}  % to address font issues with \textrightarrow
\usepackage{textcomp}                  % use better special symbols
\usepackage{mathptmx}                  % use matching math font
\usepackage{times}                     % we use Times as the main font
\renewcommand*\ttdefault{txtt}         % a nicer typewriter font
\usepackage{cite}
%% We encourage the use of mathptmx for consistent usage of times font
%% throughout the proceedings. However, if you encounter conflicts
%% with other math-related packages, you may want to disable it.

%% In preprint mode you may define your own headline.
%\preprinttext{To appear in IEEE Transactions on Visualization and Computer Graphics.}

%% If you are submitting a paper to a conference for review with a double
%% blind reviewing process, please replace the value ``0'' below with your
%% OnlineID. Otherwise, you may safely leave it at ``0''.
\onlineid{0}

%% declare the category of your paper, only shown in review mode
\vgtccategory{Research}

%% Paper title.
\title{Protein-Protein Interaction Interactive Exploration}

%% This is how authors are specified in the journal style

%% indicate IEEE Member or Student Member in form indicated below
\author{Katar\'{i}na Furmanov\'{a}, Kate\v{r}ina Kratochv\'{i}lov\'{a}, Jan By\v{s}ka, Ivan Viola, Eduard M. Gr\"{o}ller, Barbora Kozl\'{i}kov\'{a}}
\authorfooter{
%% insert punctuation at end of each item
\item
 Katar\'{i}na Furmanov\'{a} is with Masaryk University, Czech Republic. E-mail: furmanova@mail.muni.cz.
\item
 Kate\v{r}ina Kratochv\'{i}lov\'{a} is with Masaryk University, Czech Republic. E-mail: 374486@mail.muni.cz.
\item
 Jan By\v{s}ka is with Masaryk University, Czech Republic. E-mail: xbyska@fi.muni.cz.
\item 
Eduard M. Gr\"{o}ller is with TU Wien, Austria. Email: groeller@cg.tuwien.ac.at.
\item
Ivan Viola is with TU Wien, Austria. Email: viola@cg.tuwien.ac.at.
\item
Barbora Kozl\'{i}kov\'{a} is with Masaryk University, Czech Republic. E-mail: kozlikova@fi.muni.cz. 
}

%other entries to be set up for journal
\shortauthortitle{Furmanov\'{a} \MakeLowercase{\textit{et al.}}: Protein-Protein Interaction Interactive Exploration}
%\shortauthortitle{Firstauthor \MakeLowercase{\textit{et al.}}: Paper Title}

%% Abstract section.
\abstract{Studying the patterns of protein interactions is fundamental for understanding the structure and function of biological complexes. The exploration of the vast space of possible mutual conformations of interacting proteins and their contact zones is very time consuming and requires non-trivial user experience. Therefore, in this paper we propose three novel methods for guided exploration of the conformational space which help the domain experts to select the most biochemically relevant contact zones and explore them on different levels of detail. The first method, based on customized interactive heat maps, provides the overview of all possible protein conformations and their interactive filtering. The second method enables to traverse the pre-filtered conformations using a lens view. Here the conformation in focus is equipped with the information about interacting amino acids. These techniques are interactively linked with the third proposed method which represents individual conformations in three dimensional space. The problem of high overlaps of the conformations is solved by using exploded views equipped with 2D close-up views showing the details of the contact zone on different levels of abstraction. The usefulness of our methods was evaluated by the domain experts studying the structural maintenance of chromosomes.
} % end of abstract

%% Keywords that describe your work. Will show as 'Index Terms' in journal
%% please capitalize first letter and insert punctuation after last keyword
\keywords{Protein-protein interaction, heat map, exploded view, contact zone.}

%% ACM Computing Classification System (CCS). 
%% See <http://www.acm.org/class/1998/> for details.
%% The ``\CCScat'' command takes four arguments.

\CCScatlist{ % not used in journal version
 \CCScat{I.3.6}{Picture/Image Generation}%
{Line and Curve Generation};
}

%% Uncomment below to include a teaser figure.
   \teaser{
   \centering
   \includegraphics[width=16cm]{teaser.png}
   \caption{Overview of our proposed visualization methods for exploration of conformations of protein-protein interactions. (a) Matrix view aggregating the information about all conformations. (b) Lens view for exploration of individual conformations and their interacting amino acids. (c) Exploded view enabling to observe the position of individual conformations. (d) Open-book view enabling to explore the shape and constitution of the contact zones. (e) List view serving for comparison of pairing amino acids of individual conformations. TODO change image and text.}
  }

%% Uncomment below to disable the manuscript note
%\renewcommand{\manuscriptnotetxt}{}

%% Copyright space is enabled by default as required by guidelines.
%% It is disabled by the 'review' option or via the following command:
% \nocopyrightspace

\vgtcinsertpkg

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%% START OF THE PAPER %%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{document}

%% The ``\maketitle'' command must be the first command after the
%% ``\begin{document}'' command. It prepares and prints the title block.

%% the only exception to this rule is the \firstsection command
\firstsection{Introduction}

\maketitle
Understanding the constitution and biological function of proteins is essential in many research disciplines, namely in medicine and pharmaceutics.
This knowledge is tightly connected with the ability of the protein to interact with other molecules.
Proteins can interact with small molecules, called ligands, which enter the protein.
This process, called protein-ligand docking, is widely used in protein engineering where the goal is to change specific properties of a given protein by performing a chemical reaction between the protein and a ligand.
In drug design, the protein serves as a "factory" for ligand structural changes caused again by a mutual chemical reaction. 
Such modified ligand can then serve as a basis of a new drug. 
However, in drug design also the interactions between proteins are attracting increasing attention because most of the proteins critical for cellular life act in a cooperative manner forming multiprotein complexes. 
It is estimated that about 800 complexes exist in just one yeast cell. 
And all complexes are composed of subunits which constitute the complex via protein-protein interactions.
The main goal of the process of studying such protein-protein interactions, known as protein-protein docking, is to identify an appropriate spatial conformation of interacting proteins.
This requires the investigation and understanding of the mechanisms of the interactions and the identification of so called "hot spot" amino acids forming the contact zones between the interacting proteins.

Structure-determination of protein-protein interactions in laboratories is very challenging and time and money consuming, because of many problems related to the dynamic nature of the proteins, the difficulties with their purification and preparation of samples.
Therefore, computational docking helping with early understanding of the feasibility of proposed conformations between interacting proteins is currently very promising option and many algorithms and tools appeared in the last years.
The categorization of the existing algorithms along with the description of their basic principles was published recently by Huang~\cite{Huang2014}.
However, these tools are producing a large number of possible conformations and the domain expert has to explore them and select the most biochemically relevant ones. 
Therefore, the next step is to enhance the exploration process by other computational of visual support.
There were already several algorithms published for re-ranking of the conformations according to different criteria thus suggesting the user those conformations which should be explored in detail.
As a representative of these attempts, Malhotra et al.~\cite{Malhotra2015} in 2015 presented DockScore, a webserver for ranking the individual conformations produced by the docking tools. 
Their idea is based on building a scoring scheme considering several interface parameters, such as surface area, hydrophobicity, spatial clustering, etc.
This helps the user to reduce the amount of conformations to a smaller set which still has to be explored manually.
For this exploration the visual support is essential, enabling to see the spatial orientation of the contact zones and hot spot amino acids and to compare different conformations.
Figure~\ref{fig:dockscore} shows the existing visual representation of two different conformations which uses the combination of traditionally used techniques, cartoon and balls\&sticks.
The green and yellow proteins represent two different docking solutions with respect to the violet protein. 
Red and blue amino acids illustrate individual hot spot amino acids of the two corresponding contact zones.

\begin{figure}[bt]
  \centering
  \includegraphics[width=0.8\columnwidth]{dockscore.png}
  \caption{Visual representation of the native and computed conformations of the protein complex with PDB ID 1SYX. Image taken from~\cite{Malhotra2015}.}
  \label{fig:dockscore}
\end{figure}

It is clearly visible that even for the comparison of two conformations the traditional representation suffers from many occlusion problems and it is hard to perceive the differences between individual solutions.
When comparing more conformations, even without detailed visualization of hot spot amino acids, the problem becomes even more visible (see Figure~\ref{fig:problem}).

\begin{figure}[b]
  \centering
  \includegraphics[width=0.7\columnwidth]{problem.png}
  \caption{Superimposition of several possible conformations between two proteins visualized using a traditional cartoon method. The set of green protein instances corresponds to one of
the interacting proteins, the colored components represent the second protein in different conformations.}
  \label{fig:problem}
\end{figure}

Therefore, in this paper we are proposing a solution to this problem.
We present three novel methods for visualization and comparison of individual conformations which aim to remove the problems of the existing solution and provide the domain experts with an intuitive and user friendly tool for interactive exploration of protein-protein interactions proposed by the computational tools.
Our solution is designed for dealing with large number of conformations so the user is not forced to use any of the re-ranking algorithms before using our solution. 

\section{Related Work}
In this section we will review the existing techniques related to our proposed representations of protein-protein interactions.
The problem of visual representation can be taken from different views.
One group of the existing solutions focuses on the visualization of whole networks of interacting proteins.
Because of their complexity, i.e., the number of interacting proteins, the visualizations are mostly graph-based.
Jeanquartier et al.~\cite{Jeanquartier2015} presented a survey of databases enabling the visual analysis of protein networks.

Second group consists of techniques visualizing the contact zones and their interacting amino acids.
The spatial techniques have to deal with the problem of occlusion and visual clutter caused by the fact that the most interesting parts of the interacting proteins, the contact zones, are positioned close to each other (see Figure~\ref{fig:varshney}a).
In other words, without any transformation or a visual enhancement (e.g., transparency) it is impossible to visually explore the contact zones.
Therefore, Jin et al.~\cite{Jin2014} presented the open-book view where the interacting proteins are rotated to face the contact zones towards the camera.
An alternative approach presented by Lee and Varshney~\cite{Varshney2003} computes and visualizes the intermolecular negative volume and the area of the docking site (Figure~\ref{fig:varshney}b,c).
This way the users can observe the volume between the interacting proteins without direct visualization of the contact zones themselves.
It can serve the domain experts as an interactive tool for studying possible docking conformations.

\begin{figure}[tb]
  \centering
  \includegraphics[width=1.0\columnwidth]{varshney.png}
  \caption{(a) Surface visualization of two interacting proteins with PDB ID 4SGB where the contact zones are not visible. (b) and (c) Visualization of the intermolecular negative surface representing the volume between the contact zones. Image taken from~\cite{Varshney2003}.}
  \label{fig:varshney}
\end{figure}

Alternative approaches to the visualization of contact zones can use abstracted 2D representations.
As an example of such approach can be taken the schematic representation of the contact zones used in the PDBsum database~\cite{pdbsum} which uses the overview visualization representing each of the interacting proteins by a sphere equipped with the information about the number of amino acids forming the contact zones and the number of different types of interactions between them.
The second visualization lists all contact zone amino acids. 
The interactions are visualized by lines of different colors and thickness which represent the type and strength of the interaction.

\begin{figure}[bt]
  \centering
  \includegraphics[width=1.0\columnwidth]{pdbsum.png}
  \caption{Two abstracted visualizations of the complex SMC3-Scc1 with PDB ID 4UX3 available in the PDBsum database. (a) Overview representation showing the number of amino acids in the contact zones and the types on interactions. (b) Part of the list of all interacting amino acids along with individual interactions and their strength.}
  \label{fig:pdbsum}
\end{figure}

\textcolor{red}{TODO some other list views??}

\textcolor{red}{TODO matrices?? Molecollar?}

\section{Overview}
Our newly proposed visualizations aim to help the domain experts to explore a set of possible conformations detected by one of the existing computational tools and to select the most biochemically relevant ones.
The relevance consists of both geometrical fitting of the contact zones and the chemical compatibility between interacting amino acids which is derived from their physico-chemical properties.
Therefore, the users have to iteratively filter out those conformations which are not fulfilling given criteria.
To follow this workflow, summarized in Figure~\ref{fig:workflow}, we proposed a set of specific visualizations. 
The input datasets consisting of dozens of conformations between two interacting proteins were computed using HADDOCK~\cite{haddock}, one of the currently most used tool.
In the following sections we will describe our newly proposed techniques which aim to visually support and enhance the exploration workflow.

\begin{figure}[bt]
  \centering
  \includegraphics[width=1.0\columnwidth]{workflow.png}
  \caption{Overview of the process of exploration of the input conformations using our newly proposed methods.}
  \label{fig:workflow}
\end{figure}

\section{Matrix View}
When the biochemists have some preliminary knowledge about the interacting proteins and know about a pair of amino acids from these proteins that they are surely interacting, they can give this information as an input parameter directly to the computational tool.
If no such information is available, the tool searches for possible conformations randomly.
In both cases, the resulting set of conformations is large and requires some preliminary filtering.
This filtering can be based in the prior knowledge or can consider those pairs of amino acids which are present in most of the resulting conformations.
To support this task, we proposed the matrix-based visualization similar to commonly used heat maps (see Figure~\ref{fig:matrixlens} left part).
The rows and columns of the matrix correspond to two interacting proteins.
Each row or column represents one amino acid present in a contact zone of some of the conformations. 
In other words, the rows and columns are formed only by those amino acids of the interacting proteins which are in contact in at least one conformation.
The contact between the amino acids is based on their mutual Euclidean distance -- two amino acids are considered to be in contact when their distance is between 3 and 5 \AA ngstr\"{o}ms.
The color of each rectangle in the matrix corresponds to the number of the occurrences of the corresponding interacting amino acids in the set of conformations. 
The intense red color represents pairs of amino acids which are interacting in most of the conformations.
Thus, it can be anticipated that they play a significant role in the protein-protein interaction and it is meaningful to explore them in more detail.
On the other hand, rare combinations of amino acids can be filtered out.
The matrix serves directly for filtering out those improbable solutions by interactive selection of rectangles.
Moreover, the matrix allows the selection of a combination of more pairs of amino acids.
This is useful when the user wants to further explore only those conformations which contain the interactions between amino acids $A$, $B$ and/or simultaneously also $C$, $D$. 

\begin{figure}[bt]
  \centering
  \includegraphics[width=1.0\columnwidth]{matrixlens.png}
  \caption{Matrix view (left) showing the aggregated information about the presence of mutually interacting amino acids in all conformations. It is interactively linked with the lens view (right) presenting more detailed information about individual conformations and their interacting amino acids.}
  \label{fig:matrixlens}
\end{figure}

One of the biggest advantages of the matrix view is its independence on the size of the input set.
The number of row and columns is limited by the size of interacting proteins, so in the worst case it can correspond to the total number of amino acids of these proteins.
However, usually the number of amino acids in the contact zones is very limited so the number of rows and columns is much smaller than this upper limit.
Then, each conformation of the input dataset increases the counters represented by the color in the matrix rectangles.
Therefore, the only parameter influencing the size of the matrix is the size of the interacting proteins.
For large proteins the rectangles of the matrix can be too small.
In such situations, the users can use the table lens technique extended to both rows and columns of the matrix.

The matrix view is interactively linked with another newly proposed visualization, lens view, positioned directly next to the matrix (Figure~\ref{fig:matrixlens} right part). 

\section{Lens View}
This representation accompanying the matrix view aims to provide the users with more detailed information about individual conformations.
In the matrix view the information about all conformations is merged in order to give the overview of the interacting amino acids throughout the whole dataset.
Therefore, the constitution of individual conformations, yet very important in the filtering stage, is omitted.
Our lens view overcomes this limitation of the matrix view and presents the content of the contact zones of individual conformations in a highly abstracted way.

The central part of the lens view consists of rectangles representing individual conformations. 
The biggest rectangle in focus (in the center) corresponds to the currently investigated conformation. 
It contains the identifier of the conformation and a set of polylines connecting amino acids which are interacting in this conformation.
Moreover, the closest amino acids in the corresponding contact zones are highlighted (by thicker and colored polyline).
The lens view is highly interactive and closely linked with the matrix view.
The user can scroll the view with the mouse wheel and show and hide the connecting polylines.
The pairs of amino acids forming the conformation in focus can be highlighted in the matrix view so the user can immediately see the number of conformations in which these pairs are present as well.
Vice versa, by interacting with the matrix view and selecting given rectangles, the lens view can be automatically filtered and shows only those conformations which satisfy the filtering condition defined in the matrix view.

By combining the matrix and lens views, the domain expert can inspect the large input set of conformations and primarily assess their biochemical relevance.
Selected conformations can be further processed by the following visualization methods.

\section{Exploded View}
The biochemists are already well accustomed with the manipulation with molecules in three-dimensional environment, thus this space has to be an integral part of the workflow.
However, as already stated, exploring and comparing many structures in 3D at once suffers from problems with high overlaps, occlusion, and visual clutter. 
So the traditionally used representations are not sufficient (as demonstrated in Figures~\ref{fig:dockscore} and~\ref{fig:problem}).
To overcome these limitations, we use the technique of so called exploded views which enlarges the distance between the interacting proteins (see Figure~\ref{fig:exploded}). 

\begin{figure}[bt]
  \centering
  \includegraphics[width=1.0\columnwidth]{exploded.png}
  \caption{Exploded view showing the interaction between the reference protein and its paired proteins. Proteins are represented by their surface and the contact zones are colored.}
  \label{fig:exploded}
\end{figure}

The main principle is the following.
First the user selects one of the interacting proteins as the reference one.
Lets mark the second protein in this interaction as the paired protein.
All these reference proteins taken from the selected conformations are aligned using a Combinatorial Extensions structural alignment algorithm~\cite{Shindyalov1998} (see the green protein in Figure~\ref{fig:problem}).
The set of proteins interacting with the reference proteins are positioned around these reference proteins.
As it is expected that the set of conformations selected in the previous stage contains more items, the paired proteins are overlapping significantly.
Therefore, by increasing the distance between the reference and paired proteins using exploded views can remove this obstacle.
To ensure that the paired proteins in the exploded view will not collide with each other, we have to calculate their spatial position based on their bounding boxes.
To maintain the information about the position of the contact zones on both reference and paired protein, we display their connection as \textcolor{red}{TODO}.

The resulting visualization removed not only the problem of overlapping paired proteins but also enabled the user to see the contact zones which were in the original position of interacting proteins hidden between them.
However, this solution does not solve the problem that the contact zones are still faced to each other so the user has to manipulate with the camera to observe the contact zone from a perpendicular view. 
But even such manipulation does not enable to see both contact zones at once (the second contact zone is then located on the opposite side of the camera).
This problem is solved by the open-book view presented in the following section.

\section{Open-Book View}
This view is activated when the user selects one of the conformations from the exploded view. 
Then the visualization of other conformations is suppressed and the open-book view is launched.
This view consists of the animation in which the reference and paired proteins are rotated and translated so that they are positioned next to each other and their contact zones are facing towards the observer (see Figure~\ref{fig:book}). 
To maintain the information about the pairing of amino acids, the user can visualize also individual connections by simple lines.

\begin{figure}[bt]
  \centering
  \includegraphics[width=0.8\columnwidth]{book.png}
  \caption{Open-book view enabling to explore the contact zones between the interacting proteins simultaneously. The surface of the contact zones can be color-coded according to different criteria -- here the color represents the hydrophobicity of individual amino acids. \textcolor{red}{TODO change image}}
  \label{fig:book}
\end{figure}

The contact zones represented by their surface can be color-coded according to different criteria.
The color can correspond the mutual distance between the amino acids -- the closest amino acids have different color than the most distant ones.
It can also represent different physico-chemical properties of the amino acids or their atoms, such as hydrophobicity or partial charges.
To inform the users about the types and identifiers of individual amino acids, the surfaces can be equipped with labels bearing this information.

In both the exploded view and the open-book view the protein can be represented also by other traditionally used visualization methods, such as cartoon, spheres, balls\&sticks, sticks, etc.
Moreover, these methods can be combined so, for example, the proteins can be represented by surfaces and the amino acids in the contact zones can be visualized using sticks representation to see their spatial orientation. \textcolor{red}{TODO image???}

The combination of exploded view and open-book view is very useful for exploration of individual conformations.
But when the task is to compare individual conformations with respect to the pairs of interacting amino acids, these views can give only a limited information.
Therefore, we proposed another abstracted view serving mainly for the comparison of the paired amino acids in individual solutions.

\section{List View}
The list view consists of two sets of amino acids in the contact zones, each set coming from one interacting protein (see Figure~\ref{fig:list}).
The left part contains all amino acids coming from the reference protein, the right part is formed by their counterparts in the paired protein.
For each conformation one list view is created and all list views are juxtapositioned so the user can see and visually compare the constitution of the contact zones of all selected conformations.
The user can interact with this representation by changing the properties of the amino acids mapped onto their corresponding rectangles (same properties as those mapped onto the surface in the 3D views) and sort the list according to them.

\begin{figure}[bt]
  \centering
  \includegraphics[width=0.8\columnwidth]{list.png}
  \caption{\textcolor{red}{TODO description}. \textcolor{red}{TODO change image}}
  \label{fig:list}
\end{figure}

\section{User Interaction}
Except for the interaction with individual visualizations which were described along with their principles, all proposed visualizations are interactively linked to support the workflow of the biochemists as much as possible.
It means that by the selection of rectangles in the matrix view the user automatically filters out those conformations which do not fulfill the filtering criteria and these conformations are further omitted from the exploded, open-book, and list views as well.
Of course, the user can anytime return to the beginning of the process and by iterative refining of the filtering using the matrix and lens views the set of conformations can be further narrowed.
Another possible scenario is that the user reveals that some of the promising conformations was filtered out at the beginning.
Such conformations can be then returned back to the exploration process.

\section{Demonstration and Results}
All presented techniques were designed in tight cooperation with the domain experts from the field of functional genomics and proteomics.
Their current main research focus is on structure maintenance of chromosome (SMC) complexes. 
The SMC complexes are key players in chromatin organization where they ensure the proper condensation of chromosomes. 
The researchers analyze the architecture and function of such complexes using a variety of experimental approaches. 
Their goal is to uncover the way the subunits of these complexes interact with each other and execute unique function(s) of these complexes. 
Therefore, a visual representation of such information is highly beneficial because it helps to reveal the spatial relationships between the subunits in an intuitive way.

To demonstrate the usability of our proposed techniques, the domain experts selected an example of the hNSE1-hNSE3 

\section{Conclusion}

%% if specified like this the section will be committed in review mode
%\acknowledgments{
%The authors wish to thank A, B, C. This work was supported in part by
%a grant from XYZ.}

%\bibliographystyle{abbrv}
\bibliographystyle{abbrv-doi}
%\bibliographystyle{abbrv-doi-narrow}
%\bibliographystyle{abbrv-doi-hyperref}
%\bibliographystyle{abbrv-doi-hyperref-narrow}
%%use following if all content of bibtex file should be shown
%\nocite{*}
\bibliography{template}
\end{document}

